From 4a4e16d24af126bd62f7b6a3270484e9c72fa6d5 Mon Sep 17 00:00:00 2001
From: Delphine CC Chiu <Delphine_CC_Chiu@wiwynn.com>
Date: Mon, 25 Sep 2023 11:10:46 +0800
Subject: [PATCH] Add mctp eids configuration for Yosemite 4

The MCTP specification is defined in DSP0236. As describe in section
8.17.2, we need a configuration for static EIDs. The `mctpd` from
CodeConstuct/mctp will be able to know where those static EIDs located.
The other services will be able to know what those EIDs stand for by the
"Name" as well.

schemas: add `EndpointId` with type `string` and `number`.

Tested: check D-Bus path for configuration.
Change-Id: I587f5a69051f783753f5855d2f51242aa1cd5bc4
Signed-off-by: Delphine CC Chiu <Delphine_CC_Chiu@wiwynn.com>
---
 configurations/yosemite4_floatingfalls.json | 17 ++++++++++++++-
 configurations/yosemite4_sentineldome.json  | 10 ++++++++-
 configurations/yosemite4_wailuafalls.json   | 24 ++++++++++++++++++++-
 schemas/legacy.json                         |  6 ++++++
 4 files changed, 54 insertions(+), 3 deletions(-)

diff --git a/configurations/yosemite4_floatingfalls.json b/configurations/yosemite4_floatingfalls.json
index e51624f..e0075c4 100644
--- a/configurations/yosemite4_floatingfalls.json
+++ b/configurations/yosemite4_floatingfalls.json
@@ -1,5 +1,20 @@
 {
-    "Exposes": [],
+    "Exposes": [
+        {
+            "Address": "0x20",
+            "Bus": "$bus % 16",
+            "EndpointId": "$bus % 15 * 10 + 1",
+            "Name": "BIC",
+            "Type": "MCTPEndpoint"
+        },
+        {
+            "Address": "0x20",
+            "Bus": "$bus % 16",
+            "EndpointId": "$bus % 15 * 10 + 3",
+            "Name": "CXL",
+            "Type": "MCTPEndpoint"
+        }
+    ],
     "Name": "Yosemite 4 Floating Falls Slot $bus % 15",
     "Probe": "xyz.openbmc_project.FruDevice({'BOARD_PRODUCT_NAME': 'Floating Falls', 'PRODUCT_PRODUCT_NAME': 'Yosemite V4'})",
     "Type": "Board",
diff --git a/configurations/yosemite4_sentineldome.json b/configurations/yosemite4_sentineldome.json
index d18bf4c..cdfe8bd 100644
--- a/configurations/yosemite4_sentineldome.json
+++ b/configurations/yosemite4_sentineldome.json
@@ -1,5 +1,13 @@
 {
-    "Exposes": [],
+    "Exposes": [
+        {
+            "Address": "0x20",
+            "Bus": "$bus % 16",
+            "EndpointId": "$bus % 15 * 10",
+            "Name": "BIC",
+            "Type": "MCTPEndpoint"
+        }
+    ],
     "Name": "Yosemite 4 Sentinel Dome Slot $bus % 15",
     "Probe": "xyz.openbmc_project.FruDevice({'BOARD_PRODUCT_NAME': 'Sentinel Dome', 'PRODUCT_PRODUCT_NAME': 'Yosemite V4'})",
     "Type": "Board",
diff --git a/configurations/yosemite4_wailuafalls.json b/configurations/yosemite4_wailuafalls.json
index 6f8a85b..97dfae1 100644
--- a/configurations/yosemite4_wailuafalls.json
+++ b/configurations/yosemite4_wailuafalls.json
@@ -1,5 +1,27 @@
 {
-    "Exposes": [],
+    "Exposes": [
+        {
+            "Address": "0x20",
+            "Bus": "$bus % 16",
+            "EndpointId": "$bus % 15 * 10 + 2",
+            "Name": "BIC",
+            "Type": "MCTPEndpoint"
+        },
+        {
+            "Address": "0x20",
+            "Bus": "$bus % 16",
+            "EndpointId": "$bus % 15 * 10 + 4",
+            "Name": "CXL1",
+            "Type": "MCTPEndpoint"
+        },
+        {
+            "Address": "0x20",
+            "Bus": "$bus % 16",
+            "EndpointId": "$bus % 15 * 10 + 5",
+            "Name": "CXL2",
+            "Type": "MCTPEndpoint"
+        }
+    ],
     "Name": "Yosemite 4 Wailua Falls Slot $bus % 15",
     "Probe": "xyz.openbmc_project.FruDevice({'BOARD_PRODUCT_NAME': 'Wailua Falls', 'PRODUCT_PRODUCT_NAME': 'Yosemite V4'})",
     "Type": "Board",
diff --git a/schemas/legacy.json b/schemas/legacy.json
index 86dd16d..e1c259e 100644
--- a/schemas/legacy.json
+++ b/schemas/legacy.json
@@ -69,6 +69,9 @@
                 "EntityInstance": {
                     "$ref": "#/definitions/Types/EntityInstance"
                 },
+                "EndpointId": {
+                    "$ref": "#/definitions/Types/EndpointId"
+                },
                 "FaultIndex": {
                     "$ref": "#/definitions/Types/FaultIndex"
                 },
@@ -650,6 +653,9 @@
             "EntityInstance": {
                 "type": "number"
             },
+            "EndpointId": {
+                "type": ["string", "number"]
+            },
             "FaultIndex": {
                 "type": "number"
             },
-- 
2.25.1

